p4proto/kctrl/switchapi/README.md

```diff
@@ -0,0 +1,165 @@
+
+ * **switch_api_l3_route_delete**: Retrieve the route handle from hashtable using route entry as the key. Call the platform dependent(PD) API to delete the table entries from target backend and delete the route handle.
+
+## Platform Dependent(PD) API <a name="platform_dependent_api"></a>
+
+All platform dependent (PD) API's allocates a session handle for TDI transactions and retrieves the key and table handle and id's associated for action and all required fields with appropriate TDI API's and program the target backend using tdi_table_entry_add API. Once the table is successfully programed, the session handle is deleted
+
+#### FDB platform dependent API:
+ * **switch_pd_l2_tx_forward_table_entry**
+   *  For FDB learns from vlan interface
+```
+    Table name: LNW_L2_FWD_TX_TABLE
+    Key: LNW_L2_FWD_TX_TABLE_KEY_DST_MAC
+    Action: LNW_L2_FWD_TX_TABLE_ACTION_L2_FWD with parameters port id of vlan interface
+```
+   * For FDB learns from physical interface
+```
+    Table name: LNW_L2_FWD_TX_TABLE
+    Key: LNW_L2_FWD_TX_TABLE_KEY_DST_MAC
+    Action: LNW_L2_FWD_TX_TABLE_ACTION_L2_FWD with parameters port id of physical interface
+```
+   *  For FDB learns from tunnel interface
+```
+    Table name: LNW_L2_FWD_TX_TABLE
+    Key: LNW_L2_FWD_TX_TABLE_KEY_DST_MAC
+    Action: LNW_L2_FWD_TX_TABLE_ACTION_SET_TUNNEL with parameters tunnel id, dst addr
+```
+ * **switch_pd_l2_rx_forward_table_entry**
+```
+    Table name: LNW_L2_FWD_RX_TABLE
+    Key: LNW_L2_FWD_TX_TABLE_KEY_DST_MAC
+    Action: LNW_L2_FWD_RX_TABLE_ACTION_L2_FWD with parameters port id
+```
+
+#### Route Platform dependent API
+
+ * **switch_pd_nexthop_table_entry**
+```
+    Table name: LNW_NEXTHOP_TABLE
+    Key: LNW_NEXTHOP_TABLE_KEY_NEXTHOP_ID
+    Action: LNW_NEXTHOP_TABLE_ACTION_SET_NEXTHOP with parameters rif id, neighbor id and egress port id
+```
+ * **switch_pd_neighbor_table_entry**
+```
+    Table name: LNW_NEIGHBOR_MOD_TABLE
+    Key: LNW_NEIGHBOR_MOD_TABLE_KEY_VENDORMETA_MOD_DATA_PTR
+    Action: LNW_NEIGHBOR_MOD_TABLE_ACTION_SET_OUTER_MAC with parameters dst_mac_addr
+```
+
+ * **switch_pd_rif_mod_entry**
+```
+    Table name: LNW_RIF_MOD_TABLE
+    Key: LNW_RIF_MOD_TABLE_KEY_RIF_MOD_MAP_ID
+    Action: LNW_RIF_MOD_TABLE_ACTION_SET_SRC_MAC with parameters src_mac_addr
+```
+
+ * **switch_pd_ipv4_table_entry**
+```
+    Table name: LNW_IPV4_TABLE
+    Key: LNW_IPV4_TABLE_KEY_IPV4_DST_MATCH
+    For SWITCH_ACTION_NHOP:
+        Action LNW_IPV4_TABLE_ACTION_SET_NEXTHOP_ID with parameters nexthop_id
+    For SWITCH_ACTION_ECMP :
+        Action LNW_IPV4_TABLE_ACTION_ECMP_HASH_ACTION with parameters ecmp_group_id
+```
+
+ * **switch_pd_ecmp_hash_table_entry**
+```
+    Table name: LNW_ECMP_HASH_TABLE
+    Key: LNW_ECMP_HASH_TABLE_KEY_HOST_INFO_TX_EXTENDED_FLEX_0
+    Action: LNW_ECMP_HASH_TABLE_ACTION_SET_NEXTHOP_ID with parameters nexthop_id
+
+#### Tunnel Platform Dependent API
+
+ * **switch_pd_tunnel_entry**
```

Comment:
This is not aligned with previous hyperlinks in the document
eg: switch_pd_ecmp_hash_table_entry

Reply:
Fixed the alignment